plugins/reporters/fossid-snippets/src/main/kotlin/FossIdSnippetReporter.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2023 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.plugins.reporters.fossidsnippets
+
+import java.io.File
+
+import org.apache.logging.log4j.kotlin.Logging
+import org.ossreviewtoolkit.plugins.reporters.asciidoc.HtmlTemplateReporter
+import org.ossreviewtoolkit.plugins.reporters.freemarker.FreemarkerTemplateProcessor
+import org.ossreviewtoolkit.reporter.Reporter
+import org.ossreviewtoolkit.reporter.ReporterInput
+
+class FossIdSnippetReporter : Reporter {
+    companion object : Logging {
+
+        private const val TEMPLATE_NAME = "fossid_snippets"
+
+        val delegateReporter = HtmlTemplateReporter()
```

Comment:
Should we consider opening `HtmlTemplateReporter` and use inheritance instead? Or Kotlin built-in class delegation?

Reply:
`Reporter` has only one function `generateReport`. So if we use Kotlin built-in class delegation, we would nevertheless override the function to be able to set the option, nullifying the benefits of the class delegation.
Shall we open the `HtmlTemplateReporter ` instead ?